### ISSUE TITLE
Standard libs Math and polyfills (charCodeAt, fromCharCode, etc)

### ISSUE THREAD

**Issue by MaxGraey** [01alchemist/TurboScript#14]
According to this: https://github.com/WebAssembly/design/blob/master/FAQ.md we need manually implement all sets of js enviroment. This is big job. I could help with this

**Comment by MaxGraey** [comment]
How you bind Math library to Wasm right now? It seems just import from js according this: https://github.com/01alchemist/TurboScript/blob/master/tests/math-wasm.html.

One more thing. In declared set of WasmOpcodes is missing f32_fma and f64_fma: https://github.com/01alchemist/TurboScript/blob/master/src/wasm/opcode.ts

**Comment by nidin** [comment]
Right now importing from JavaScript built-in. I need to implement method overload to use same functions for different signatures like in JavaScript. It's a big task to implement all built-ins manually. We need to wait for WASM 2 Specification to be finalized before implementing this manually. For now, for compatibility we can import slow Math library from JS and gradually add fast implementations in WASM.

**Comment by MaxGraey** [comment]
That's right we definitely need fma command for IEEE-754 (2008) compatible sin, cos, log and etc, but min, max, sqrt, rounding and abs we already has direct equivalents in wasm, so we can use them now

**Comment by nidin** [comment]
Sure we can implement direct equalents right away. right now built-in classes and functions are located inside library folder.

**Comment by winksaville** [comment]
Be sure there are no associated costs for modules that don't use these features.

**Comment by nidin** [comment]
Tree shaking should remove unused functions.
So there will be no extra cost.

**Comment by MaxGraey** [comment]
@nidin Is it possible use wasm's sqrt, clz, ctz, popcount, copysign and etc now?

**Comment by nidin** [comment]
Not yet. No work done in that direction.
You are free to investigate.
